exp/instancestate/awsinstancestate_controller.go

```diff
@@ -94,7 +95,7 @@ func (r *AwsInstanceStateReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	// retrieve queue URL if it isn't already tracked
 	if _, ok := r.queueURLs.Load(awsCluster.Name); !ok {
 		URL, err := r.getQueueURL(awsCluster)
-		if err != nil {
+		if err != nil && !queueNotFoundError(err) {
```

Comment:
I think we show queueNotFoundError, only when feature is enabled like this:
`	if feature.Gates.Enabled(feature.EventBridgeInstanceState) {}
`
So that users that enable it, sees this error and understand what is wrong.

Reply:
we are setting this feature gate as [true](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/data/e2e_conf.yaml#L212) for E2E tests, so this log flooding will keep on existing in E2E artifacts logs

Reply:
Do we actually test the event bridge functionality in the e2e tests? Also, what creates the SQS queue in the first place (i.e is it auto-created by AWS or is it something that we need to create manually).

I agree with @sedefsavas i think we need to display the error here if the queue doesn't exist but we need to work out why the queue doesn't exist and fix that as the root cause.

Reply:
I see it's enabled [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/test/e2e/shared/template.go#L96)
Not sure where it is used.

Reply:
Is it ok if we disable this feature flag?

Reply:
We could disable the event bridge feature if we aren't testing it in the e2e and that would stop the messages. However, i'm not sure that truly gets to the bottom of what is causing the error.

I'm curious to know why we are getting this error if the event bridge feature is enabled and CAPA is responsible for creating the queues? This implies there is a bug somewhere in either the e2e tests or the reconciliation loop.

This is a weird part of the error as well: `The specified queue does not exist for this wsdl version.`

Reply:
Based on my brief investigation during e2e runs, it seems that `AWS.SimpleQueueService.NonExistentQueue` error occurs because the awsinstancestate_controller tries to retrieve a queue url before a queue is created and after a queue is deleted. 

I think there is a bug in the reconciliation loop that needs to be fixed.

Reply:
@Ankitasw let's see if this is actually a bug and if so, we can do the fix in this PR. 
Also, we are missing documentation for this feature https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2978, in case you are interested.

Reply:
@sedefsavas @richardcase I agree to what @pydctw is telling, I checked the logs, that are populating even before queue is created. I think we should check for NonExistentQueue error [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/instancestate/awsinstancestate_controller.go#L95-L101) for reconcile loop.

Reply:
One issue with not logging error in case queue is not found is if AWSCluster controller fails to create queue, user won't know why this feature is not working.

We can add a condition with information severity here:

https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/3e61428d885f2eaeab1a7dffc4d17270a20c6675/controllers/awscluster_controller.go#L238-L245

Reply:
I think if queue is not created successfully, it would be captured [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/pkg/cloud/services/instancestate/queue.go#L36-L48)

Reply:
👍 let us know if the pr is ready.

Reply:
it's ready now.